From c0cc22d2621105f3d7d9791c801dee31929b686c Mon Sep 17 00:00:00 2001
From: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Date: Thu, 6 Dec 2018 18:15:39 +0900
Subject: [PATCH 055/909] pinctrl: sh-pfc: r8a77990: Disable USB30_OVC pin
 pullup/pulldown control

commit 951aa269fa1637aba9e951496361c44eca94a538 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

According to the R-Car Gen3 Hardware Manual Errata for Rev 1.00 of
Aug 24, 2018, changing the USB30_OVC pin from the pullup state of the
initial value was prohibited.

This patch disables USB30_OVC pin pullup/pulldown control.

Signed-off-by: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/pinctrl/sh-pfc/pfc-r8a77990.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/pinctrl/sh-pfc/pfc-r8a77990.c b/drivers/pinctrl/sh-pfc/pfc-r8a77990.c
index 97becb2955a9..9a5d56c79927 100644
--- a/drivers/pinctrl/sh-pfc/pfc-r8a77990.c
+++ b/drivers/pinctrl/sh-pfc/pfc-r8a77990.c
@@ -3455,7 +3455,9 @@ static void r8a77990_pinmux_set_bias(struct sh_pfc *pfc, unsigned int pin,
 	if (bias == PIN_CONFIG_BIAS_PULL_UP)
 		updown |= BIT(bit);
 
-	sh_pfc_write(pfc, reg->pud, updown);
+	if (pin != RCAR_GP_PIN(6, 9))
+		sh_pfc_write(pfc, reg->pud, updown);
+
 	sh_pfc_write(pfc, reg->puen, enable);
 }
 
-- 
2.17.1

